src/test/java/com/netflix/playback/features/model/impl/PeriodAggregatorTest.java

```diff
@@ -0,0 +1,68 @@
+package com.netflix.playback.features.model.impl;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+
+import com.netflix.playback.features.model.impl.PeriodAggregator;
+
+import junit.framework.Assert;
+
+public class PeriodAggregatorTest {
+
+	@Test
+	public void testAggregateData() throws InterruptedException {
+		PeriodAggregator aggregator = new PeriodAggregator();
+		ConcurrentHashMap<Integer, AtomicInteger> testMap = new ConcurrentHashMap<>();
+		testMap.put(1, new AtomicInteger(4));
+		testMap.put(2, new AtomicInteger(3));
+		aggregator.flush(testMap, 1);
+		Thread.sleep(1000);
```

Comment:
instead of using aggregator.flush() to inject the data, we can add method updatePeriodToEventCount() in PeriodAggregator which updates the periodToEventCount object and in tests we could remove Thread.sleep()